### PR TITLE
Add fastify-xml-body-parser in community list

### PR DESCRIPTION
**Pull request by amitguptagwl**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jsumners** [comment]
This seems wrong:

> This plugin use fastify-xml-body-parser to parse the XML payload. So it accepts all the options supported by fastify-xml-body-parser.

How does the plugin use itself to parse XML?

**Comment by amitguptagwl** [comment]
Ah! thanks to highlight. I've updated the README now.

**Comment by amitguptagwl** [comment]
Adding error handler will be a good idea. But can there be any benefit to limit body size?

**Comment by delvedor** [comment]
> But can there be any benefit to limit body size?

https://snyk.io/vuln/npm:fastify:20180107

**Comment by amitguptagwl** [comment]
Hmm, Thanks. I never thought in that way. I'll make it configurable

**Comment by amitguptagwl** [comment]
body limit can be more effective when it is done on stream instead of reading the complete request body first. Hence IMHO, a framework should provide that option but a user should put such check on API gateway layer.

**Comment by delvedor** [comment]
We have updated our content type parser api, I suggest you to give a read to the [documentation](https://github.com/fastify/fastify/blob/master/docs/ContentTypeParser.md).
[Here](https://github.com/fastify/fastify/blob/master/lib/ContentTypeParser.js#L115-L131) you can see how we handle the body size.

**Comment by amitguptagwl** [comment]
Now it's look perfect. One question though.

> In this way, all of the requests that do not have a corresponding content type parser will be handled by the specified function.

Does it mean custom parser handling specific type will be called on priority than parser with `*`?

**Comment by nwoltman** [comment]
@amitguptagwl Yes, the `*` parser will only be used if no other custom parser matches the request.

**Comment by StarpTech** [comment]
@amitguptagwl please resolve the conflict after this we can merge it.

File: docs/Ecosystem.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ Plugins maintained by the fastify team are listed under [Core](#core) while plug
 - [`fastify-datastore`](https://github.com/now-ims/now-fastify-datastore) Fastify plugin for [Google Cloud Datastore](https://cloud.google.com/nodejs/docs/reference/datastore/1.4.x/).
 - [`fastify-dynamodb`](https://github.com/matrus2/fastify-dynamodb) AWS DynamoDB plugin for Fastify. It exposes [AWS.DynamoDB.DocumentClient()](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html) object.
 - [`fastify-error-page`](https://github.com/hemerajs/fastify-error-page) Fastify plugin to print errors in structured HTML to the browser.
+- [`fastify-xml-body-parser`](https://github.com/NaturalIntelligence/fastify-xml-body-parser) Parse XML payload / request body into JS / JSON object.
```
Comment (mcollina) [inline docs/Ecosystem.md]: Can you sort this alphabetically?

**Comment by amitguptagwl** [comment]
Hello Team! (with due respect of your work and time)

Fastify is one of the popular web framework. Multiple commits on this file is expected. This PR was raised a long time back. Hence I'll appreciate if we can review to merge/close it.

Thanks @mcollina for your reviews.

**Comment by mcollina** [comment]
Sorry for the long delay!

**Comment by amitguptagwl** [comment]
@mcollina @delvedor Thanks to merge it. I really appreciate your effort to make this wonderful opensource web framework. I hope my another PR in other repo will be reviewed soon. :smile:

**Pull request merged**